### PR TITLE
Bad resource links will no longer kill Module construction.

### LINKED ISSUE #1208: Bad links will cause netkan.exe to cry.
{Kerbal Foundries](http://www.kerbalstuff.com/mod/58) has a source code link of `http://included%20in%20the%20download`. This isn't a valid `System.Uri` string, which causes `Module` inflation to fail, which causes `netkan.exe` to bail.

We should log and continue if a resources URL is bad, rather than bailing entirely. I have a fix on the way.

### PR DESCRIPTION
**Pull request by pjf**
Fixes #1208, which is preventing some mods from indexing correctly.

### CODE REVIEW

**Comment by pjf** [comment]
Found a spammy bug, one moment while I fix. :)

**Comment by pjf** [comment]
And less spammy now. :) Review/merges welcome. :)

File: Core/Converters/JsonIgnoreBadUrlConverter.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+            if (value == null)
+                return null;
+
+            Uri url;
+
+            try
+            {
+                url = new Uri(value);
+                return url;
```
Comment (RichardLake) [inline Core/Converters/JsonIgnoreBadUrlConverter.cs]: return new Uri(value);
Also why have the declaration outside the try when the catch/following code doesn't use it.

Comment (pjf) [inline Core/Converters/JsonIgnoreBadUrlConverter.cs]: Because I wrote this when I was very tired. :) Thanks for catching these! :)

**Comment by pjf** [comment]
@RichardLake : Code tidy-ups based upon your feedback is in. :)

**Pull request merged**